splink/aws/aws_linker.py

```diff
@@ -0,0 +1,186 @@
+        t = self.get_schema_info(self.physical_name)
+        d = wr.catalog.get_table_types(
+            database=t[0],
+            table=t[1]
+        )
+
+        return list(d.keys())
+
+    def validate(self):
+        pass
+    
+    def drop_table_from_database(self, force_non_splink_table=False):
+        
+        self._check_drop_folder_created_by_splink(force_non_splink_table)
+        self._check_drop_table_created_by_splink(force_non_splink_table)
+        self.aws_linker.drop_table_from_database_if_exists(self.physical_name)
+        self.aws_linker.delete_table_from_s3(self.physical_name)
+    
+    def _check_drop_folder_created_by_splink(self, force_non_splink_table=False):
+        
+        filepath = self.aws_linker.boto_utils.s3_output
+        filepath = filepath.split("/")[-3:-1]
+        # validate that the write path is valid
+        valid_path = [
+            self.aws_linker.boto_utils.s3_output_name_prefix, 
+            self.aws_linker.boto_utils.session_id
+            ] == filepath
+        if not valid_path:
+            if not force_non_splink_table:
+                raise ValueError(
+                    f"You've asked to drop data housed under the filepath "
+                    f"{self.aws_linker.boto_utils.s3_output} "
+                    "from your s3 output bucket, which is not a folder created by Splink. "
+                    "If you really want to delete this data, you can do so by setting "
+                    "force_non_splink_table=True."
+                )
+        
+        # validate that the ctas_query_info is for the given table we're interacting with
+        if self.aws_linker.ctas_query_info[self.physical_name]["ctas_table"] != self.physical_name:
+            raise ValueError(
+                    f"The recorded metadata for {self.physical_name} that you're "
+                    "attempting to delete does not match the recorded metadata on s3. "
+                    "To prevent any tables becoming corrupted on s3, this run will be "
+                    "terminated. Please retry the link/dedupe job and report the issue "
+                    "if this error persists."
+                )
+
+    def as_record_dict(self, limit=None):
+        sql = f"""
+        select *
+        from {self.physical_name}
+        """
+        if limit:
+            sql += f" limit {limit}"
+
+        out_df = wr.athena.read_sql_query(
+            sql=sql,
+            database=self.aws_linker.output_schema,
+            s3_output=self.aws_linker.boto_utils.s3_output,
+            keep_files=False,
+        )
+        return out_df.to_dict(orient="records")
+
+    def get_schema_info(self, input_table):
+        t = input_table.split(".")
+        return t if len(t) > 1 else [self.aws_linker.output_schema, self.physical_name]
+    
+
+class AWSLinker(Linker):
+    def __init__(self, settings_dict: dict,
+                 boto3_session: boto3.session.Session,
+                 output_schema: str,
```

Comment:
Is the correct terminology here (in the world of Athena) a `database`?  

If so I think for clarity for the end user the argument here should be called `output_database`.  `output_schema` can then be used internally e.g. `self.output_schema = output_database`.

Reply:
Yeah, that's fine. I changed it to `schema_output` to better align with the `duckdb` side of things.

Athena is a bit of a mess when it comes to what they actually mean by the data prefix (i.e. `splink.`) - the data engineers all refer to it as a schema, while amazon themselves seem to call it a database... in reality, it's not really either 🤷